src/dbsql.py

```diff
@@ -123,14 +129,34 @@ def _create_alert_api_call(self, query_id: str, alert: Alert, parent_folder: str
             rearm=alert.rearm,
         )
 
-    def _create_alert_schedule_api_call(self, alert: Alert, alert_id: str, destination_id: str, warehouse_id: str):
+    def _create_alert_schedule_api_call(self, alert: Alert, alert_id: str, destination_id: str, warehouse_id: str, run_as: str | None = None, tags: dict[str, str] = None):
         """
         Creates an alert schedule in Databricks
         """
+
+        if run_as:
+            if '@' in run_as:
+                run_as_obj = JobRunAs(user_name=run_as)
+            else:
+                run_as_obj = JobRunAs(service_principal_name=run_as)
+        else:
+            run_as_obj = None
```

Comment:
Consider extracting this logic out, since the concept is independent of alert creation. 
Eg in other languages, you'd have a factory method or multiple constructors that create instances of `JobRunAs` based on `run_as` string. That way it can be used in other places as well.